DataFormats/PatCandidates/interface/PackedCandidate.h

```diff
@@ -664,13 +652,26 @@ namespace pat {
     reco::VertexRef::key_type pvRefKey_;
 
     /// IP covariance	
-    CMS_THREAD_GUARD(vertex_) mutable float dxydxy_, dzdz_, dxydz_,dlambdadz_,dphidxy_,dptdpt_,detadeta_,dphidphi_;
+    CMS_THREAD_GUARD(vertex_) mutable reco::TrackBase::CovarianceMatrix m_;
```

Comment:
Here is the writeup we have on making DataFormats thread safe, specifically on using atomic pointers:
https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedThreadSafeDataStructures#Lazy_caching_of_a_pointer

So you create an object using new, fill it in and only once it is complete do you assign it to the atomic pointer. That way no code outside of the function doing the filling ever sees the object being filled. Either they see an empty pointer or they see the pointer pointing to a complete object.